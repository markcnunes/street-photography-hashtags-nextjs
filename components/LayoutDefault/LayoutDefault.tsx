import styled from '@emotion/styled';
import { darken, rgba } from 'polished';
import React from 'react';

interface ILayoutDefault {
  /**
   * Takes string
   */
  title?: string;
  /**
   * Content of component
   */
  children?: React.ReactNode;
  /**
   * Takes string
   */
  className?: string;
}

const StyledLayoutDefault = styled('div')(({ theme }) => ({
  flex: '1 0 0',
  color: theme.colors.white,
  a: {
    color: theme.colors.white,
  },
  p: {
    lineHeight: 1.5,
  },

  '.subheader': {
    display: 'flex',
    padding: 20,
    border: 0,
    outline: 0,
    width: '100%',
    maxWidth: '90%',
    margin: '0 auto',
    color: theme.colors.white,
    '.wrapper': {
      position: 'relative',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      textAlign: 'center',
      width: '100%',
      padding: 20,
      background: `linear-gradient(10deg, ${rgba(
        theme.colors.white,
        0.1,
      )} 0%, ${rgba(darken(0.1, theme.colors.white), 0)} 100%)`,
      borderRadius: 5,
    },
    h3: {
      margin: '0 auto',
      textTransform: 'uppercase',
    },

    '.title': {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      textAlign: 'center',
      svg: {
        marginRight: 5,
      },
    },
  },
  main: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexWrap: 'wrap',
    width: '100%',
    maxWidth: '90%',
    padding: 20,
    margin: '0 auto',

    '& > *': {
      width: '100%',
    },
  },
}));

const LayoutDefault: React.FC<ILayoutDefault> = ({ title, children }) => {
  return (
    <StyledLayoutDefault>
      <div className='subheader'>
        <div className='wrapper'>
          <h2 className='title'>{title}</h2>
        </div>
      </div>
      <main>{children}</main>
    </StyledLayoutDefault>
  );
};

export default LayoutDefault;
