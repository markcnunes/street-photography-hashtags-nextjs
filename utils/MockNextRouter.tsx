import { RouterContext } from 'next/dist/next-server/lib/router-context';
import { NextRouter } from 'next/router';
import { ReactNode } from 'react';

interface MockNextRouterProps extends Partial<NextRouter> {
  children?: ReactNode;
}

/**
 * Override the default NextRouter render with different attributes.
 * e.g. change the path or locale for tests.
 */

const MockNextRouter = ({ children, ...props }: MockNextRouterProps) => {
  const mockRouter: NextRouter = {
    route: '/',
    pathname: '',
    query: {},
    asPath: '/',
    basePath: '',
    locale: 'en',
    locales: [''],
    defaultLocale: 'en',
    // domainLocales?: DomainLocales,
    isLocaleDomain: true,
    push: jest.fn(() => Promise.resolve(true)),
    replace: jest.fn(() => Promise.resolve(true)),
    reload: jest.fn(() => Promise.resolve(true)),
    back: jest.fn(() => Promise.resolve(true)),
    prefetch: jest.fn(),
    beforePopState: jest.fn(() => Promise.resolve(true)),
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
    isFallback: false,
    isReady: true,
    isPreview: false,
  };

  return (
    <RouterContext.Provider value={{ ...mockRouter, ...props }}>
      {children}
    </RouterContext.Provider>
  );
};

export default MockNextRouter;
