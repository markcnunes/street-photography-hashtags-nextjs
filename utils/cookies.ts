// import ReactGA from 'react-ga';

import GA4React from 'ga-4-react';

export const cookieConsent = 'sphCookieconsent';

/*
 * Check if cookie consentment is true before initializing and rendering it
 */
export const getCookie = (key: string): string | undefined => {
  var b = document.cookie.match('(^|;)\\s*' + key + '\\s*=\\s*([^;]+)');
  return b ? b.pop() : '';
};

export const gaCode = 'G-W7QKGLQ9YV';
const initializeCookies = () => {
  if (!window.location.href.includes('localhost')) {
    const ga4react = new GA4React(gaCode);
    ga4react.initialize();
    // reload page to get the track running
    window.location.reload();
  }
};

export const renderCookies = (): any => {
  getCookie(cookieConsent) === 'true' && initializeCookies();
};

export default initializeCookies;
