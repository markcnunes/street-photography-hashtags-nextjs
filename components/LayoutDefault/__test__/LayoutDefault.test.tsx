import React from 'react';
import { render } from 'utils/test-utils';
import LayoutDefault from '../LayoutDefault';

describe('<LayoutDefault />', () => {
  it('matchs snapshot', () => {
    const { asFragment } = render(<LayoutDefault />);
    expect(asFragment()).toMatchSnapshot();
  });
});
