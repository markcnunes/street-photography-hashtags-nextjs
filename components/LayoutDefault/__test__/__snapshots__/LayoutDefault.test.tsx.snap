// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutDefault /> matchs snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  color: #fff;
}

.emotion-0 a {
  color: #fff;
}

.emotion-0 p {
  line-height: 1.5;
}

.emotion-0 .subheader {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 20px;
  border: 0;
  outline: 0;
  width: 100%;
  max-width: 90%;
  margin: 0 auto;
  color: #fff;
}

.emotion-0 .subheader .wrapper {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  width: 100%;
  padding: 20px;
  background: linear-gradient(10deg, rgba(255,255,255,0.1) 0%, rgba(230,230,230,0) 100%);
  border-radius: 5px;
}

.emotion-0 .subheader h3 {
  margin: 0 auto;
  text-transform: uppercase;
}

.emotion-0 .subheader .title {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.emotion-0 .subheader .title svg {
  margin-right: 5px;
}

.emotion-0 main {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  max-width: 90%;
  padding: 20px;
  margin: 0 auto;
}

.emotion-0 main>* {
  width: 100%;
}

<div
    class="emotion-0 emotion-1"
  >
    <div
      class="subheader"
    >
      <div
        class="wrapper"
      >
        <h2
          class="title"
        />
      </div>
    </div>
    <main />
  </div>
</DocumentFragment>
`;
